src/carte_table_to_graph.py

```diff
@@ -13,315 +13,265 @@
             )
 
+        # Ensure numerical columns exist before fitting the transformer
+        if self.num_col_names:
+            num_cols_exist = [col for col in self.num_col_names if col in X.columns]
+            if num_cols_exist:
+                self.num_transformer_.fit(X[num_cols_exist])
+
+        self.is_fitted_ = True
         return self
 
     def transform(self, X, y=None):
-        """Apply Table2GraphTransformer to each row of the data
+        """
+        Apply Table2GraphTransformer to each row of the data.
 
         Parameters
         ----------
-        X : Pandas DataFrame. (n_samples, n_features)
-            The input data used to transform to graphs.
-
-        y : None
-            Ignored.
+        X : pandas.DataFrame
+            Input data to transform.
+        y : array-like, optional
+            Target values, by default None.
 
         Returns
         -------
-        Graph Data : list of size (n_samples).
-            The list of transformed graph data.
+        data_graph : list
+            List of transformed graph objects.
         """
-
-        # Preprocess the features
-        X_ = X.copy()
-        X_ = X_.replace("\n", " ", regex=True)
+        X_ = X.replace("\n", " ", regex=True)
         num_data = X_.shape[0]
 
-        # Preprocess the target
-        y_ = None
-        if self.y_ is not None:
-            y_ = np.array(self.y_)
-            y_ = torch.tensor(y_).reshape((num_data, 1))
+        y_ = torch.tensor(self.y_, dtype=torch.float32).reshape((num_data, 1)) if self.y_ is not None else None
 
-        # Separate categorical and numerical columns
         X_categorical = X_.select_dtypes(include="object").copy()
         X_categorical.columns = self.cat_col_names
         X_numerical = X_.select_dtypes(exclude="object").copy()
         X_numerical.columns = self.num_col_names
 
-        # Features for names
-        cat_names = pd.melt(X_categorical)["value"]
-        cat_names = cat_names.dropna()
-        cat_names = cat_names.astype(str)
-        cat_names = cat_names.str.replace("\n", " ", regex=True).str.lower()
-        cat_names = cat_names.unique()
-        names_total = np.hstack([self.col_names, cat_names])
-        names_total = np.unique(names_total)
-        name_dict = {names_total[i]: i for i in range(names_total.shape[0])}
-
-        # preprocess values
+        cat_names = pd.melt(X_categorical)["value"].dropna().astype(str).str.lower().unique()
+        names_total = np.unique(np.hstack([self.col_names, cat_names]))
+        name_dict = {name: idx for idx, name in enumerate(names_total)}
+
         name_attr_total = self._transform_names(names_total)
-        if len(self.num_col_names) != 0:
-            X_numerical = self._transform_numerical(X_numerical)
-        if self.is_fitted_ == False:
-            self.is_fitted_ = True
+        if self.num_col_names:
+            num_cols_exist = [col for col in self.num_col_names if col in X.columns]
```

Comment:
the num_col_exist does not contain the ABV variable (for the wina_pl example), causing different results. I think there is a mix-up between the lower- and upper-cases.